core/client/fs/src/main/java/alluxio/client/block/stream/GrpcBlockingStream.java

```diff
@@ -77,7 +78,9 @@
   public GrpcBlockingStream(Function<StreamObserver<ResT>, StreamObserver<ReqT>> rpcFunc,
       int bufferSize, String description) {
     LOG.debug("Opening stream ({})", description);
-    mResponses = new ArrayBlockingQueue<>(bufferSize);
+    // Use an unlimited queue to avoid blocking the network threads. Depend on custom flow
+    // control to limit the size of buffer.
+    mResponses = new LinkedBlockingQueue<>();
```

Comment:
Is the default buffer size (16) not large enough? With 2MB message size we are looking at 32MB buffer per reader, this can grow quickly if multi-thread reading is used.

Reply:
I originally saw this grow to around 20-25 before.

I changed it back, but is it a bad thing to block the grpc thread? Because that will sometimes happen if the size is 16.

Reply:
I think when a grpc thread is blocked it will potentially result in new thread being created for handling other concurrent RPC calls, so there is a potential overhead of one blocking thread per stream.